cryptoimpl/SSLImpl.cpp

```diff
@@ -56,16 +60,26 @@ SSLImpl::SSLImpl(ACE_HANDLE sock, const char *pubkeyfile,
 
     SSL_CTX_set_cipher_list(sslctx->context(), "DEFAULT");
     sslctx->set_mode(ACE_SSL_Context::SSLv23_client);
-    sslctx->load_trusted_ca(pubkeyfile);
+    if (sslctx->load_trusted_ca(pubkeyfile) != 0) {
+      throw Exception("Failed to read trust store.");
+    }
 
     if (strlen(password) > 0) {
       SSL_CTX_set_default_passwd_cb(sslctx->context(), pem_passwd_cb);
       SSL_CTX_set_default_passwd_cb_userdata(sslctx->context(),
                                              const_cast<char *>(password));
     }
 
-    sslctx->private_key(privkeyfile);
-    sslctx->certificate(privkeyfile);
+    if (sslctx->private_key(privkeyfile) != 0) {
+      throw Exception("Failed to read private key.");
```

Comment:
There ought to be a better Exception class to throw here.

Reply:
Added new SSLException type and instantiated with appropriate exception strings.